xcmds/cpu/init.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This is init code for the case that cpu finds itself as pid 1.
+// This is duplicative of the real init, but we're implementing it
+// as a duplicate so we can get some idea of:
+// what an init package should have
+// what an init interface should have
+// So we take a bit of duplication now to better understand these
+// things. We also assume for now this is a busybox environment.
+// It is unusual (I guess?) for cpu to be an init in anything else.
+// So far, the case for an init pkg is not as strong as I thought
+// it might be.
```

Comment:
it can work that way, but to reduce the number of moving parts people have to think about I decided to make it work as an init too. It just made things a big more concentrated.